### PR TITLE
Solutions for sorting algorithm challenges in Coding Interview Prep/Algorithms

### LINKED ISSUE #35814: Need solutions for sorting algorithm challenges in Coding Interview Prep/Algorithms
**Describe your problem and how to reproduce it:**
Located in the English coding challenges, under Coding Interview Prep, under Algorithms, the sorting algorithms have no solutions.

**Add a Link to the page with the problem:**
https://github.com/freeCodeCamp/freeCodeCamp/tree/master/curriculum/challenges/english/08-coding-interview-prep/algorithms 

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version: 
* Operating System: Mac OS

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by wang1998tina**
- [x ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x ] My pull request targets the `master` branch of freeCodeCamp.
- [x ] None of my changes are plagiarized from another source without proper attribution.
- [x ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x ] My changes do not use shortened URLs or affiliate links.

This was a joint effort with @rns350, @dnntrn, @LifeVirus

Closes #35814

### CODE REVIEW

**Comment by RandellDawson** [comment]
Don't get rid of the following lines in the challenge seed code.
```
  // change code below this line

  // change code above this line
```

File: curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-selection-sort.english.md [hunk 1]
```diff
@@ -41,14 +41,11 @@ tests:
 
 ```js
 function selectionSort(array) {
-  // change code below this line
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-selection-sort.english.md]: Please do not remove `  // change code below this line` from the challenge seed code.

File: curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-selection-sort.english.md [hunk 2]
```diff
@@ -41,14 +41,11 @@ tests:
...
-
-  // change code above this line
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-selection-sort.english.md]: Please do not remove `  // change code above this line` from the challenge seed code.

File: curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-insertion-sort.english.md [hunk 3]
```diff
@@ -41,14 +41,10 @@ tests:
 
 ```js
 function insertionSort(array) {
-  // change code below this line
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-insertion-sort.english.md]: Please do not remove `// change code below this line` from the challenge seed code.

File: curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-insertion-sort.english.md [hunk 4]
```diff
@@ -41,14 +41,10 @@ tests:
...
-
-  // change code above this line
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-insertion-sort.english.md]: Please do not `remove // change code above this line` from the challenge seed code.

File: curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-bubble-sort.english.md [hunk 5]
```diff
@@ -43,14 +43,10 @@ tests:
 
 ```js
 function bubbleSort(array) {
-  // change code below this line
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-bubble-sort.english.md]: Please do not remove `// change code below this line` from the challenge seed code.

File: curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-bubble-sort.english.md [hunk 6]
```diff
@@ -43,14 +43,10 @@ tests:
...
-
-  // change code above this line
```
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/algorithms/implement-bubble-sort.english.md]: Please do not remove `// change code above this line` from the challenge seed code.

**Review by RandellDawson — DISMISSED** [review]
@wang1998tina, @LifeVirus, @dnntrn, @rns350 In addition to the request to not remove the commented lines I specify below, you need to rewrite your solution for the Quick Sort, as we want to be able to pass a single argument (the array to be sorted).  This way, the tests do not have to be rewritten and the user can still write an iterative solution if they want. 

The other four sorting algorithm solutions appear fine and I would be fine merging just those 4 (once you put the comment lines back in).  To speed up the process of getting at least the 4 solutions merged, why don't you just revert the changes to the `implement-quick-sort.english.md` file on this PR and then create a new PR which only adds a solution.  In the new PR you should not change any of the challenge seed code or tests.

Thank you.

**Review by RandellDawson — APPROVED** [review]
LGTM

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by moT01 — APPROVED** [review]
LGTM @wang1998tina - thanks for contributing to freeCodeCamp! 🎉 

Thanks for commiting changes @RandellDawson 🎉

**Pull request merged**